src/rust/engine/process_execution/src/nailgun/mod.rs

```diff
@@ -302,27 +258,21 @@ impl CapturedWorkdir for CommandRunner {
     let nails_command = self
       .async_semaphore
       .with_acquired(move || {
-        Self::materialize_workdir_for_server(
-          store,
-          workdir_for_this_nailgun.clone(),
-          jdk_home,
-          req.input_files,
-          workunit_store,
-        )
-        .and_then(move |_metadata| {
-          // Connect to a running nailgun.
-          executor.spawn_on_io_pool(futures::future::lazy(move || {
-            nailgun_pool.connect(
-              nailgun_name.clone(),
-              nailgun_req,
-              &workdir_for_this_nailgun1,
-              nailgun_req_digest,
-              build_id,
-            )
-          }))
-        })
-        .map_err(|e| format!("Failed to connect to nailgun! {}", e))
+        // Connect to a running nailgun.
+        executor.spawn_on_io_pool(futures::future::lazy(move || {
```

Comment:
Now that connect returns a future you shouldn't need to wrap it in a futures::lazy call.